WooCommerce/Classes/ViewRelated/Products/Media/Media+WPMediaAsset.swift

```diff
@@ -2,37 +2,39 @@ import MobileCoreServices
 import WPMediaPicker
 import Yosemite
 
-extension Media: WPMediaAsset {
+extension CancellableMedia: WPMediaAsset {
     public func image(with size: CGSize, completionHandler: @escaping WPMediaImageBlock) -> WPMediaRequestID {
-        let imageURL = thumbnailURL ?? src
+        let imageURL = media.thumbnailURL ?? media.src
         guard let url = URL(string: imageURL) else {
             return 0
         }
 
         // TODO-2073: move image fetching to `WordPressMediaLibraryPickerDataSource` instead of having to use the `ServiceLocator` singleton on a `Media`
         // extension.
         let imageService = ServiceLocator.imageService
-        imageService.retrieveImageFromCache(with: url) { (image) in
+        imageService.retrieveImageFromCache(with: url) { [weak self] (image) in
             if let image = image {
                 completionHandler(image, nil)
                 return
             }
 
-            imageService.downloadImage(with: url, shouldCacheImage: true) { (image, error) in
+            self?.cancellableTask = imageService.downloadImage(with: url, shouldCacheImage: true) { (image, error) in
```

Comment:
I believe this should be:
```suggestion
            self?.cancellableTask = imageService.downloadImage(with: url, shouldCacheImage: true) { [weak self] (image, error) in
```

Reply:
hmm I don't see the use of `self` in the completion block, and `completionHandler` in the completion block isn't mutated? lemme know if I missed anything!

Reply:
You are right, `self` is not used within the block. Please ignore what I said 😄